src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java

```diff
@@ -249,10 +255,10 @@ public void drawScreen(final int mouseX, final int mouseY, final float btn) {
 
         this.tooltip = -1;
 
-        final int offY = 23;
+        final int offY = SECTION_HEIGHT;
         int y = 0;
         int x = 0;
-        for (int z = 0; z <= 4 * 5; z++) {
+        for (int z = 0; z <= ITEMS_PER_ROW * rows; z++) {
```

Comment:
perhaps magic 2 in line 274 shouldbe changed to (ITEMS_PER_ROW-1) then

Reply:
reworked. There are some other bugs was present here